src/ape/contracts/base.py

```diff
@@ -126,6 +126,48 @@ def __call__(self, *args, **kwargs) -> Any:
             address=self.contract.address,
         )(*args, **kwargs)
 
+    def as_transaction(self, *args, **kwargs):
+        """
+        Convert the call to a transaction. This is useful for checking coverage
+        or checking gas costs.
+
+        Args:
+            *args: The contract method invocation arguments.
+            **kwargs: Transaction kwargs, such as value or
+              sender.
+
+        Returns:
+            :class:`~ape.api.transactions.TransactionAPI`
+        """
+        args = self._convert_tuple(args)
+        selected_abi = _select_method_abi(self.abis, args)
+
+        contract_transaction = ContractTransaction(  # type: ignore
+            abi=selected_abi,
+            address=self.contract.address,
+        )
+
+        transaction = contract_transaction.serialize_transaction(*args, **kwargs)
+        self.provider.prepare_transaction(transaction)
+        return transaction
+
+    def estimate_gas_cost(self, *args, **kwargs) -> int:
+        """
+        Get the estimated gas cost (according to the provider) for the
+        contract method call (as if it were a transaction).
+
+        Args:
+            *args: The contract method invocation arguments.
+            **kwargs: Transaction kwargs, such as value or
+              sender.
+
+        Returns:
+            int: The estimated cost of gas to execute the transaction
+            reported in the fee-currency's smallest unit, e.g. Wei.
+        """
+        txn = self.as_transaction(*args, **kwargs)
+        return txn.max_fee if txn.max_fee else self.provider.estimate_gas_cost(txn)
```

Comment:
Hmm after more thinking, we should not do that check here. Let's return `self.provider.estimate_gas_cost(txn)` and let plugins handling specific cases.

Reply:
reverted!